src/engine/test/source/builder/builders/opBuilderHelperSHAfrom_test.cpp

```diff
@@ -0,0 +1,161 @@
+/* Copyright (C) 2015-2022, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <any>
+#include <gtest/gtest.h>
+#include <vector>
+
+#include <baseTypes.hpp>
+
+#include "opBuilderSHAfrom.hpp"
+
+using namespace base;
+namespace bld = builder::internals::builders;
+
+// SHA1 hash results
+constexpr char wordWorldWorstHash[] {"6184625af204bbfc7e751d49cdc1d7c7ee15091a"};
+constexpr char xxxHash[] {"a9674b19f8c56f785c91a555d0a144522bb318e6"};
+constexpr char wordHash[] {"3cbcd90adc4b192a87a625850b7f231caddf0eb3"};
+constexpr char abcdeHash[] {"7be07aaf460d593a323d0db33da05b64bfdcb3a5"};
+
+TEST(opBuilderSHAfrom, Builds)
+{
+    auto tuple = std::make_tuple(std::string {"/field"},
```

Comment:
Please, add the `const` modifier to every variable that won't change its value. If you have values that are repeated many times you could define a `constexpr` or `const string`, or similar, to avoid redundancy of code.

Reply:
done